localstack/services/s3/provider.py

```diff
@@ -494,6 +494,41 @@ def copy_object(
             src_moto_bucket, key=src_key, version_id=src_version_id
         )
 
+        # see https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html
+        condition = None
+        source_object_last_modified = source_key_object.last_modified.replace(
+            tzinfo=ZoneInfo("GMT")
```

Comment:
i wanted to suggest to use `datetime.timezone.utc` for this, as i think this is what we want. i can see we're using `ZoneInfo("GMT")` in other areas in the s3 codebase though. maybe we can take a note to unify this with the rest of the codebase, where we tend to use `datetime` or `dateutil`.

Reply:
actually i can see that parts of S3 really operate on GTM. in that case: disregard :-)

Reply:
Yes, most headers from S3 really returns GMT, I think it's the format for these headers (RFC1123 maybe?). I like ZoneInfo as it's now baked it since 3.9, and I think it's the recommended way to get timezone info now?

Reply:
Also, I need to do this because moto is storing them as TZ unaware I believe, whereas the serializer/parser correctly parse the datetime, which is great